src/main/java/io/vertx/core/json/Json.java

```diff
@@ -239,12 +243,36 @@ public void serialize(Instant value, JsonGenerator jgen, SerializerProvider prov
     }
   }
 
+  private static class InstantDeserializer extends JsonDeserializer<Instant> {
+    @Override
+    public Instant deserialize(JsonParser p, DeserializationContext ctxt) throws IOException, JsonProcessingException {
+      String text = p.getText();
+      try {
+        return Instant.from(ISO_INSTANT.parse(text));
+      } catch (DateTimeException e) {
+        throw new InvalidFormatException(p, "Expected an ISO 8601 formatted date time", text, Instant.class);
+      }
+    }
+  }
+
   private static class ByteArraySerializer extends JsonSerializer<byte[]> {
-    private final Base64.Encoder BASE64 = Base64.getEncoder();
 
     @Override
     public void serialize(byte[] value, JsonGenerator jgen, SerializerProvider provider) throws IOException {
-      jgen.writeString(BASE64.encodeToString(value));
+      jgen.writeString(Base64.getEncoder().encodeToString(value));
+    }
+  }
+
+  private static class ByteArrayDeserializer extends JsonDeserializer<byte[]> {
+
+    @Override
+    public byte[] deserialize(JsonParser p, DeserializationContext ctxt) throws IOException, JsonProcessingException {
+      String text = p.getText();
+      try {
+        return Base64.getDecoder().decode(text);
+      } catch (IllegalArgumentException e) {
+        throw new InvalidFormatException(p, "Expected a base64 encoded byte array", text, Instant.class);
```

Comment:
Why Instant.class ?

Reply:
This is what the Jackson `InvalidFormatException` expects. It helps give the user better failure messages.